massgov/mayflower-react
# DP-9253 fix dropdown not closing on blur, set focus on Esc.

Testing:
- open the TypeAheadDropdown story http://localhost:6006/?selectedKind=molecules&selectedStory=TypeAheadDropdown
- click or tab-focus+space|enter to toggle open
- click outside the window, or down in the knobs area -- the panel should close fully now
- open panel & re-click the button, panel should close
- open panel & press Enter or Esc. or Tab -- panel should close
-- bonus new feature: if you use Esc. the button will regain keyboard focus
-- interacting with the panel dropdown to select an org should all still be normal
